martin/src/srv/styles.rs

```diff
@@ -0,0 +1,39 @@
+use crate::styles::StyleSources;
+use actix_web::http::header::ContentType;
+use actix_web::middleware;
+use actix_web::web::{Data, Path};
+use actix_web::{route, HttpRequest, HttpResponse};
+use serde::Deserialize;
+
+#[derive(Deserialize, Debug)]
+struct StyleRequest {
+    style_id: String,
+}
+
+#[route(
+    "/style/{style_id}",
+    method = "GET",
+    wrap = "middleware::Compress::default()"
+)]
+#[allow(clippy::unused_async)]
+async fn get_style_json(
```

Comment:
currently, this does not ensure that the style is valid json or compress the whitespace (other than general compression).

Unsure if adding is better to streaming the file out via `actix_files`

Reply:
Compacting json on the fyl sounds like a good idea, especially if we enable caching (separate PR please, caching is messy). Streaming might be a good option - I wonder if we can just hand a stream to actix?

Reply:
Not sure if caching is something that we need..
local file reads are cached by the os anyway

=> I would not expect much more performance

A bit, but not much..

Reply:
Caching for this seems a bit overengineered.
Lets remember that if we access the disk often, the file system caches this too.

Caching the compressed output would be fair gain though..

Streaming was a nice idea, but does not work given that `middleware::Compress::default` does need to buffer.
I have pushed this and etags or last_modified header handling to a future PR.